tools/isobuild/package-api.js

```diff
@@ -339,13 +339,39 @@ _.extend(PackageAPI.prototype, {
     var errors = [];
     _.each(paths, function (path) {
       forAllMatchingArchs(arch, function (a) {
+        const isAsset = !! (fileOptions && fileOptions.isAsset);
+
+        // Check if we have already added a file at this path
         if (_.has(self.sources[a], path)) {
-          errors.push("Duplicate source file: " + path);
+          const otherSource = self.sources[a][path];
+
+          // Files can actually be added twice for each architecture: once as
+          // an asset, and once as a source file
+          const otherFileOptions = otherSource.fileOptions;
+          const otherFileIsAsset = !! (otherFileOptions &&
```

Comment:
So what happens on the *third* call to addFiles?  This will get treated right here as "true"

Maybe if either source or otherSource already has isAsset: both (either because the user actually typed that, or if it was corrected by the last invocation of this code and we're on the third call now) it's an error?

ie, you can either literally pass "isAsset: both", or you can write two lines with isAsset true and isAsset false, but you can't do anything more confusing than that.

Alternatively, would it not be simpler to just tell people who want to do this weird thing to write `isAsset: 'both'` instead of implementing this extra special case?  Yes, that means people will have to update their package.js, but we're already telling lots of people to update their package.js for assets in this release.  Asking a very small number of people to write "both" seems simpler than maintaining this 20 lines of special cased exception to the "one mention of a file" rule.